proposals/0000-symbol-newtype-string.rst

```diff
@@ -0,0 +1,128 @@
+  type FS = Fields '[ "foo", "bar" ]  -- ok
+
+As with ``Nat`` and ``Natural`` proposal, it's possible to go roundabout
+this (and e.g. ``singletons-base`` does). However, if ``Symbol`` were
+an ordinary ``newtype`` over ``String``, we would avoid this issue entirely.
+
+(Note: ``singletons-base`` demotes ``Symbol`` to ``Text`` which is
+incorrect, as ``Text`` cannot represent all ``String`` values)
+
+Proposed Change Specification
+-----------------------------
+
+* Change ``data Symbol`` to ``newtype Symbol = MkSymbol String``
+
+In particular, we don't propose any changes to ``GHC.TypeLits`` interface.
+Firstly the implementation of ``Symbol`` stays internal.
+Also ``symbolVal`` will continue to return ``String`` value.
+In future the ``GHC.Symbol`` interface mirroring ``GHC.TypeLits`` functionlity
+could be vreated , but using ``Symbol`` also for the term level can be made as
+CLC proposal, as that would be only a library change.
+
+Technically this is a library only change, but maybe it's only
+a happy coincidence that GHC internals don't need to be changed,
+when ``Symbol`` definition is changed.
+
+Effect and Interactions
+-----------------------
+
+This change is completely backwards-compatible.
+
+Costs and Drawbacks
+-------------------
+
+This proposal introduces "yet another string" type into Haskell ecosystem.
+Yet, it serves as specific purpose.
+
+Alternatives
+------------
+
+An alternative is to export ``Symbol`` definition from ``GHC.TypeLits``.
+This however raises questions like whether ``MkSymbol '[] ~ ""``.
+For now we pretend that ``MkSymbol`` should not be promoted,
+until that can be resticted.
+
+It is also possible to prevent ``MkSymbol`` usage on the type level by
+defining ``Symbol`` like::
+
+  data Symbol = MkSymbol (# String #)
+
+As unlifted types cannot be promoted, it won't be possible to successfully
+use ``MkSymbol`` on the type level. (No more success than with ``NS`` and
+``NB`` constructors of type ``Natural``).A
+
+An alternative is to make ``String`` promote to ``Symbol``.
+That won't work out well, because lists of characters (i.e. ``String``)
+can be promoted already.
+
+The opposite alternative is to make ``type Symbol = String``.
+It was also meentioned as an alternative in The Char Kind proposal.
+It was rejected then, and the reasoning is still valid:
+First of all, we keep ``Symbol`` for type-checking efficiency.
+Moreover, we would also handle type families inside cons cells when solving
+``HasField`` constraints. For example, ``HasField T ('x' : F y : G z) ty``.
+
+It's possible to use some other type for ``Symbol`` "backend" than ``String``,
+however ``Text`` doesn't qualify as ``String`` (and type-level ``Symbol``)
+can contain values ``Text`` cannot represent. For example single surrogate codepoints.
+``SNat`` and ``KnownSymbol`` are backed by ``String``, so it is a natural
+choice.
+
+Unresolved Questions
+--------------------
+
+None.
+
+Implementation Plan
```

Comment:
If `newtype Symbol = MkSymbol String` is to be mentioned it, it should only be in this section.